GrailDiary/GrailDiary/View Controllers/AddPOIViewController.swift

```diff
@@ -0,0 +1,99 @@
+//
+//  AddPOIViewController.swift
+//  GrailDiary
+//
+//  Created by Mike Nichols on 5/14/20.
+//  Copyright © 2020 Mike Nichols. All rights reserved.
+//
+
+import UIKit
+
+// #13
+protocol AddPOIDelegate {
+    func poiWasAdded(_ poi: POI)
+}
+
+class AddPOIViewController: UIViewController {
+    
+    // #12
+    
+    @IBOutlet weak var location: UITextField!
+    @IBOutlet weak var country: UITextField!
+    @IBOutlet weak var clue1: UITextField!
+    @IBOutlet weak var clue2: UITextField!
+    @IBOutlet weak var clue3: UITextField!
+    
+    // #14
+    
+    var delegate: AddPOIDelegate
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        // Do any additional setup after loading the view.
+    }
+    
+    
+    // #15
+    
+    @IBAction func cancelTapped(_ sender: UIBarButtonItem) {
+        
+        // #16
+        
+        dismiss(animated: true, completion: nil)
+    }
+    
+    @IBAction func saveTapped(_ sender: UIBarButtonItem) {
+        
+        // #17
+        
+        guard let location = location.text,
+            let country = country.text else { return }
+        
+        var clues: [String] = []
+        
+        if let clue1 = clue1.text,
+            clue1 != "" {
```

Comment:
Another good way to check is a string is empty is by writing !clue1.isEmpty

Reply:
ahhhhhh!